### PR TITLE
Keep rc before freeing it during labels unlink alarms

### LINKED ISSUE #13284: [Bug]: service cant start
### Bug description

netdata agent service cant start on client

Ubuntu 20.04.4 LTS
Linux 5.4.0-109-generic
netdata v1.35.0-104-nightly

### Expected behavior

N/A

### Steps to reproduce

1. 

### Screenshots

_No response_

### Error Logs

kernel: [4484994.051211] netdata[2946070]: segfault at 318 ip 000055acc6f89cf4 sp 00007ffd77bd7310 error 4 in netdata[55acc6e20000+359000]
 kernel: [4484994.051216] Code: 55 4c 8d 2d e6 84 22 00 41 54 49 89 f4 55 48 89 fd 53 48 89 f3 48 83 ec 08 eb 0f 0f 1f 00 48 8b 9b e0 01 00 00 48 85 db 74 75 <48> 83 bb 28 01 00 00 00 74 ea 48 8b b3 30 01 00 00 48 8b bd 28 03
 systemd[1]: netdata.service: Main process exited, code=dumped, status=11/SEGV
 systemd[1]: netdata.service: Failed with result 'core-dump'.
 systemd[1]: Created slice User Slice of UID 0.
 systemd[1]: Starting User Runtime Directory /run/user/0...
 systemd[1]: Finished User Runtime Directory /run/user/0.
 systemd[1]: Starting User Manager for UID 0...

### Desktop

Browser Version [e.g. 22]

### Additional context

_No response_

### LINKED ISSUE #13304: [Bug]: netdata crashes in `rrdcalc_labels_unlink_alarm_loop()`
### Bug description

Netdata crashes on start in `rrdcalc_labels_unlink_alarm_loop()` when health [host_labels](https://learn.netdata.cloud/docs/agent/health/reference#alarm-line-host-labels) line contains a  label name/value that the host has no.

- coredump

```c
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from /opt/netdata/usr/sbin/netdata...
[New LWP 934401]
[New LWP 934402]
[New LWP 934533]
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
Core was generated by `/opt/netdata/usr/sbin/netdata -P /opt/netdata/var/run/netdata/netdata.pid -D'.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  rrdcalc_labels_unlink_alarm_loop (host=host@entry=0x560c4ac267d0, alarms=0x560c4ac8c920) at database/rrdcalc.c:661
661	        if (!rc->host_labels) continue;
[Current thread is 1 (Thread 0x7f2c98e07580 (LWP 934401))]
(gdb) bt
#0  rrdcalc_labels_unlink_alarm_loop (host=host@entry=0x560c4ac267d0, alarms=0x560c4ac8c920) at database/rrdcalc.c:661
#1  0x0000560c479e9894 in rrdcalc_labels_unlink_alarm_from_host (host=host@entry=0x560c4ac267d0) at database/rrdcalc.c:678
#2  0x0000560c479d782f in health_reload_host (host=host@entry=0x560c4ac267d0) at health/health.c:195
#3  0x0000560c479d790d in health_reload () at health/health.c:228
#4  0x0000560c479ef42f in reload_host_labels () at database/rrdhost.c:1344
#5  0x0000560c47997880 in main (argc=<optimized out>, argv=<optimized out>) at daemon/main.c:1450
```

- valgrind report

```c
==985223== Invalid read of size 8
==985223==    at 0x17AEC8: rrdcalc_labels_unlink_alarm_loop (rrdcalc.c:660)
==985223==    by 0x17AF59: rrdcalc_labels_unlink_alarm_from_host (rrdcalc.c:678)
==985223==    by 0x168E75: health_reload_host (health.c:195)
==985223==    by 0x168F53: health_reload (health.c:228)
==985223==    by 0x180193: reload_host_labels (rrdhost.c:1182)
==985223==    by 0x1298B5: main (main.c:1457)
==985223==  Address 0x620cf00 is 480 bytes inside a block of size 488 free'd
==985223==    at 0x48399AB: free (vg_replace_malloc.c:538)
==985223==    by 0x14031B: freez (libnetdata.c:166)
==985223==    by 0x17AC84: rrdcalc_free (rrdcalc.c:597)
==985223==    by 0x17AD58: rrdcalc_unlink_and_free (rrdcalc.c:638)
==985223==    by 0x17AEC7: rrdcalc_labels_unlink_alarm_loop (rrdcalc.c:670)
==985223==    by 0x17AF59: rrdcalc_labels_unlink_alarm_from_host (rrdcalc.c:678)
==985223==    by 0x168E75: health_reload_host (health.c:195)
==985223==    by 0x168F53: health_reload (health.c:228)
==985223==    by 0x180193: reload_host_labels (rrdhost.c:1182)
==985223==    by 0x1298B5: main (main.c:1457)
==985223==  Block was alloc'd at
==985223==    at 0x483AB65: calloc (vg_replace_malloc.c:760)
==985223==    by 0x140363: callocz (libnetdata.c:176)
==985223==    by 0x16C09C: health_readfile (health_config.c:686)
==985223==    by 0x141755: recursive_config_double_dir_load (libnetdata.c:1451)
==985223==    by 0x16EE55: health_readdir (health_config.c:1273)
==985223==    by 0x168E6D: health_reload_host (health.c:192)
==985223==    by 0x168F53: health_reload (health.c:228)
==985223==    by 0x180193: reload_host_labels (rrdhost.c:1182)
==985223==    by 0x1298B5: main (main.c:1457)
==985223==
```

### Expected behavior

Netdata doesn't crash.

### Steps to reproduce

1. Change https://github.com/netdata/netdata/blob/dd01f896b466ed8411d58a816db8de2b940f7265/health/health.d/ram.conf#L45 to `true`
2. Restart netdata.service

### Installation method

kickstart.sh

### System info

```shell
n/a
```

### Netdata build info

```shell
netdata v1.35.0-132-gdd01f896b
```

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by MrZammler**
##### Summary

Fixes #13284
Fixes #13304 

Keeps `rc` in another pointer, and uses that after freeing `rc`.

##### Test Plan

Can be tested on a non k8s cluster. Just change the label of `oom_kill` alarm from `_is_k8s_node = false` to `_is_k8s_node = true`. Using master it should crash. Using this PR it should not.

Checking count of `curl 'http://localhost:19999/api/v1/alarms?all' | grep config_hash_id | wc -l` should be -1 when `_is_k8s_node = true`.

Also tested with single alert `oom_kill`. It should correctly remove the single alert and leave the agent with 0 alerts.

##### Additional Information

<details> <summary>For users: How does this change affect me?</summary>
  
</details>

### CODE REVIEW

**Review by vlvkobal — APPROVED** [review]

File: database/rrdcalc.c [hunk 1]
```diff
@@ -666,10 +666,16 @@ static void rrdcalc_labels_unlink_alarm_loop(RRDHOST *host, RRDCALC *alarms) {
                  host->hostname,
                  rc->host_labels);
 
+            RRDCALC *rc_old = rc->next;
```
Comment (vlvkobal) [inline database/rrdcalc.c:669]: It looks like you are skipping rrdcalc structures - rc->next is assigned twice.

Comment (vlvkobal) [reply database/rrdcalc.c:669]: Maybe it's better to rename `rc_old`->`rc_next` and remove `rc = rc->next` from `for(;;)`.

Comment (MrZammler) [reply database/rrdcalc.c]: Thank you @vlvkobal , indeed one rc was skipped after freeing one.

**Comment by MrZammler** [comment]
Thanks!

**Pull request merged**